code/game/atoms_movable.dm

```diff
@@ -396,6 +397,20 @@
 	AM.set_glide_size(glide_size_override)
 	return step(AM, newdir)
 
+//We're changing zlevel
+/atom/movable/proc/onTransitZ(old_z, new_z)
+	SEND_SIGNAL(src, COMSIG_MOVABLE_Z_CHANGED, old_z, new_z)
+	for(var/atom/movable/AM in src) // Notify contents of Z-transition. This can be overridden IF we know the items contents do not care.
+		AM.onTransitZ(old_z,new_z)
```

Comment:
```suggestion
/atom/movable/proc/onTransitZ(old_z, new_z)//uncomment when something is receiving this signal
	/*SEND_SIGNAL(src, COMSIG_MOVABLE_Z_CHANGED, old_z, new_z)
	for(var/atom/movable/AM in src) // Notify contents of Z-transition. This can be overridden IF we know the items contents do not care.
		AM.onTransitZ(old_z,new_z)*/ 
```